File: sample/src/main/java/com/nguyenhoanglam/sample/MainFragment.java
package com.nguyenhoanglam.sample;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import com.nguyenhoanglam.imagepicker.model.Config;
import com.nguyenhoanglam.imagepicker.model.Image;
import com.nguyenhoanglam.imagepicker.ui.imagepicker.ImagePicker;

import java.util.ArrayList;

/**
 * Created by hoanglam on 8/28/17.
 */

public class MainFragment extends Fragment {

    public static final String EXTRA_CONFIG = "Config";


    private RecyclerView recyclerView;
    private Button pickImageButton;

    private Config config;
    ;
    private ImageAdapter adapter;
    private ArrayList<Image> images = new ArrayList<>();


    public static MainFragment newInstance(Config config) {
        Bundle args = new Bundle();
        args.putParcelable(EXTRA_CONFIG, config);

        MainFragment fragment = new MainFragment();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        config = getArguments().getParcelable(EXTRA_CONFIG);
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_main, container, false);
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        recyclerView = view.findViewById(R.id.recyclerView);
        pickImageButton = view.findViewById(R.id.button_pick_image);

        pickImageButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                start();
            }
        });

        adapter = new ImageAdapter(getActivity());
        RecyclerView.LayoutManager layoutManager = new LinearLayoutManager(getActivity(), LinearLayoutManager.HORIZONTAL, false);
        recyclerView.setLayoutManager(layoutManager);
        recyclerView.setAdapter(adapter);
    }

    private void start() {

        ImagePicker.with(this)
                .setFolderMode(config.isFolderMode())
                .setCameraOnly(config.isCameraOnly())
                .setFolderTitle(config.getFolderTitle())
                .setMultipleMode(config.isMultipleMode())
                .setSelectedImages(config.getSelectedImages())
                .setMaxSize(config.getMaxSize())
                .start();
    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent data) {
        if (requestCode == Config.RC_PICK_IMAGES && resultCode == Activity.RESULT_OK && data != null) {
            images = data.getParcelableArrayListExtra(Config.EXTRA_IMAGES);
            adapter.setData(images);
        }
    }
}

File: example/src/main/java/com/imagepicker/example/MainFragment.kt
/*
 * Copyright (C) 2023 Image Picker
 * Author: Nguyen Hoang Lam <hoanglamvn90@gmail.com>
 */
package com.imagepicker.example

import android.os.Build
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import com.imagepicker.example.databinding.FragmentMainBinding
import com.nguyenhoanglam.imagepicker.model.ImagePickerConfig
import com.nguyenhoanglam.imagepicker.ui.imagepicker.registerImagePicker

class MainFragment : Fragment() {

    private var config: ImagePickerConfig? = null
    private var adapter: ImageAdapter? = null

    private var _binding: FragmentMainBinding? = null
    private val binding get() = _binding!!

    private val launcher = registerImagePicker {
        config?.selectedImages = it
        adapter!!.setData(it)
    }

    companion object {
        const val EXTRA_CONFIG = "Config"
        fun newInstance(config: ImagePickerConfig?): MainFragment {
            val fragment = MainFragment()
            val args = Bundle()
            args.putParcelable(EXTRA_CONFIG, config)
            fragment.arguments = args
            return fragment
        }
    }

    @Suppress("DEPRECATION")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        config = if (Build.VERSION.SDK_INT >= 33)
            requireArguments().getParcelable(
                EXTRA_CONFIG,
                ImagePickerConfig::class.java
            ) else requireArguments().getParcelable(EXTRA_CONFIG)
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = FragmentMainBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        adapter = ImageAdapter(requireActivity())
        binding.recyclerView.layoutManager =
            LinearLayoutManager(activity, LinearLayoutManager.HORIZONTAL, false)
        binding.recyclerView.adapter = adapter

        binding.launchPickerButton.setOnClickListener { start() }
    }

    private fun start() {
        launcher.launch(config!!)
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}